telemetry/class-tracks.php

```diff
@@ -51,6 +59,34 @@ public function __construct( string $event_prefix = 'vip_', array $global_event_
 		$this->global_event_properties = $global_event_properties;
 		$client                      ??= new Tracks_Client();
 		$this->queue                   = $queue ?? new Telemetry_Event_Queue( $client );
+
+		$this->pass_tracks_props_to_js();
+	}
+
+	/**
+	 * Passes the base properties for a track event to JavaScript.
+	 */
+	public function pass_tracks_props_to_js(): void {
+		if ( self::$has_track_props_passed_to_js ) {
+			return;
+		}
+
+		add_action( 'admin_head', function () {
+			?>
+			<script type="text/javascript"> var VIP_TRACKS_BASE_PROPS = <?php echo wp_json_encode( $this->get_tracks_props() ); ?>; </script>
```

Comment:
Rather than pollute globals, can we find a way to register these props on the Tracks library itself? If not, can we leave this as a PHP binding and leave loading as the responsibility of the implementor (us)? Many plugins already have a global that they can use, or they may want to use a server-side approach